### ISSUE TITLE
Snackbar - Announce Snackbar for accessibility

### ISSUE THREAD

**Issue by TimRoe** [department-of-veterans-affairs/va-mobile-library#416]
## Description 
Split from #414 to handle the edge case of announcing the Snackbar for the screen reader not working correctly. Ostensibly [announceForAccessibility](https://reactnative.dev/docs/accessibilityinfo#announceforaccessibility) should be able to do this, but putting in a useEffect in the component was not working as intended--it'd only announce for accessibility if multiple Snackbars were opened and would occur when dismissing one of the later ones instead of on open. This potentially entails having the logic for announcing the Snackbar at the SnackbarProvider level (always "rendered") while only triggering it when the Snackbar display appears.

## Acceptance Criteria
  

-  [x] Explore other options for getting `announceForAccessibility` to work correctly
-  [x] Clean up lingering commented out logic around announceForAccessibility

**Accessibility Requirements**
- [x] Snackbar should be announced on appearing, but not focused
